server/src/main/java/org/opensearch/search/sandbox/SandboxedRequestTrackerService.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandbox;
+
+import org.opensearch.core.tasks.resourcetracker.TaskResourceUsage;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+import org.opensearch.tasks.TaskManager;
+import org.opensearch.tasks.TaskResourceTrackingService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.LongSupplier;
+import java.util.stream.Collectors;
+
+/**
+ * This class tracks requests per sandboxes
+ */
+public class SandboxedRequestTrackerService implements TaskManager.TaskEventListeners, SandboxResourceTracker, SandboxStatsReporter {
+
+    private static final String CPU = "CPU";
+    private static final String JVM_ALLOCATIONS = "JVM_Allocations";
+    private static final int numberOfAvailableProcessors = Runtime.getRuntime().availableProcessors();
+    private static final long totalAvailableJvmMemory = Runtime.getRuntime().totalMemory();
+    private final LongSupplier timeNanosSupplier;
+    /**
+     * Sandbox ids which are marked for deletion in between the @link SandboxService runs
+     */
+    private List<String> toDeleteSandboxes;
+    /**
+     * It is used to track the task to sandbox mapping which will be useful to remove it from the @link tasksPerSandbox
+     */
+    private final TaskManager taskManager;
+    private final TaskResourceTrackingService taskResourceTrackingService;
+
+    public SandboxedRequestTrackerService(
+        TaskManager taskManager,
+        TaskResourceTrackingService taskResourceTrackingService,
+        LongSupplier timeNanosSupplier
+    ) {
+        this.taskManager = taskManager;
+        this.taskResourceTrackingService = taskResourceTrackingService;
+        toDeleteSandboxes = Collections.synchronizedList(new ArrayList<>());
+        this.timeNanosSupplier = timeNanosSupplier;
+    }
+
+    @Override
+    public void updateSandboxResourceUsages() {
+
+    }
+
+    @Override
+    public SandboxStatsHolder getStats() {
+        return null;
+    }
+
+    private AbsoluteResourceUsage calculateAbsoluteResourceUsageFor(Task task) {
```

Comment:
Are we always able to access this information on all distributions?

Reply:
Yes, This will give us approximately correct resource usage for a task.